slang-checks/src/main/java/com/sonarsource/slang/checks/CollapsibleIfStatementsCheck.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.checks;
+
+import com.sonarsource.slang.api.BlockTree;
+import com.sonarsource.slang.api.IfTree;
+import com.sonarsource.slang.api.TextRange;
+import com.sonarsource.slang.api.Tree;
+import com.sonarsource.slang.checks.api.InitContext;
+import com.sonarsource.slang.checks.api.SecondaryLocation;
+import com.sonarsource.slang.checks.api.SlangCheck;
+import java.util.Optional;
+import org.sonar.check.Rule;
+
+@Rule(key = "S1066")
+public class CollapsibleIfStatementsCheck implements SlangCheck {
+  private static final String MESSAGE = "Merge this if statement with the nested one.";
+  private static final String SECONDARY_MESSAGE = "Nested \"if\" statement";
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(IfTree.class, (ctx, ifTreeStatement) -> Optional.of(ifTreeStatement)
```

Comment:
Is there a special reason you used an `Optional` here?
It's not wrong, but it seems strange to me to use it like this on an object we know exists...

Reply:
I like to have this fluent api notation, but I agree it's a bit weird in this case. :)
I will remove the `Optional.of` wrapper.